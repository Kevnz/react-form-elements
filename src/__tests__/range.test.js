import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import Range from '../range'

describe('Range component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(<Range label="My Range Label" />)

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Range Label')
      expect(label.textContent).toBe('My Range Label')
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in', () => {
      const { getByLabelText, container } = render(
        <Range label="My Label" initialValue="23" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Label')
      expect(input.value).toBe('23')
    })
    it('should update a value when changed in', () => {
      const { getByLabelText, container } = render(
        <Range label="My Label" initialValue="23" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Label')
      fireEvent.change(input, { target: { value: 43 } })
      expect(input.value).toBe('43')
    })
  })
})
