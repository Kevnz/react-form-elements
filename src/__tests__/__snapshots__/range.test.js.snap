// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My Range Label
  </label>
  <input
    id="id-1-1"
    max="100"
    min="0"
    step="1"
    type="range"
    value="50"
  />
  <span>
    50
  </span>
</div>
`;

exports[`Range component handling input should set a value when passed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Label
  </label>
  <input
    id="id-2-1"
    max="100"
    min="0"
    step="1"
    type="range"
    value="23"
  />
  <span>
    23
  </span>
</div>
`;

exports[`Range component handling input should update a value when changed in 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-3-1"
  >
    My Label
  </label>
  <input
    id="id-3-1"
    max="100"
    min="0"
    step="1"
    type="range"
    value="23"
  />
  <span>
    23
  </span>
</div>
`;
